R/generate_subsample.R

```diff
@@ -10,45 +11,75 @@
 #' * If this rule contradicts the “less than 20” rule (above), this rule should be prioritized. For example, if we receive a sample set from a given site and event where Bins A, B, C, D, and E are each represented by five samples (total sample size = 25), process all of the samples for that site/event.
 #' * Subsampling should be random.
 #' @param con connection to the database
+#' @param sampling_event A numeric value for the sampling event you are processing (i.e. 1:10)
 #' @param season The season for which you want to pull subsamples. A season consists of all sampling events
 #' from the given year up to September 30th and from the previous year after October 1st.
 #' @examples
 #' # connect with database
 #' con <- gr_db_connect()
 #'
 #' # subsample from all sampling events in season 2024
-#' subsamples <- generate_subsample(con, 2024)
-#' @returns A named list containing a `results` table and a `summary` table.
-#' The `results` table contains all samples selected by the process with the following columns:
+#' subsamples <- generate_subsample(con, 1, 2024)
+#' @returns A named list containing a `subsample_for_sherlock` table, a `subsample_summary` table,
+#' and a `remainders_for_gt_seq` table.
+#' The `subsample_for_sherlock` table contains all samples selected by the process with the following columns:
 #'
 #' * **sample_id**
 #' * **datetime_collected**
 #' * **stream_name**
 #' * **sample_bin_code**
 #' * **sample_event_number**
+#' * **scenario**
 #'
-#' The `summary` table contains a short summary summarizing the number of samples for a given subsampling scenario by stream,
+#' The `subsample_summary` table contains a short summary summarizing the number of samples for a given subsampling scenario by stream,
 #' event, and bin. It contains the following columns:
 #'
 #' * **stream**
 #' * **event**
 #' * **bin**
 #' * **scenario**
 #' * **subsamples**
+#'
+#' The `remainders_for_gt_seq` table contains all samples **not** selected by the subsampling process with the following columns:
+#'
+#' * **sample_id**
+#' * **datetime_collected**
+#' * **stream_name**
+#' * **sample_bin_code**
+#' * **sample_event_number**
+#' * **scenario**
 #' @export
 #' @md
-generate_subsample <- function(con, season) {
+generate_subsample <- function(con, sampling_event, season) {
+
   if(!is.numeric(season)) {
     cli::cli_abort("Season must be a numeric value.")
   }
   if(nchar(season) != 4) {
     cli::cli_abort("Season must be in the format YYYY (i.e. 2024).")
   }
-  # get all sample event IDs for a season
+
+  # get samples from the right season
   samples <- grunID::sample_filter_to_season(con, season) |>
+    # filter to sampling event of interest
+    dplyr::filter(sample_event_number == sampling_event) |>
+    # now make sure that they have the right status
+    dplyr::left_join(dplyr::tbl(con, "sample_status") |>
+                       dplyr::collect() |>
```

Comment:
no need to collect() here let the join happen on the database and collect() at the end